packages/components/src/util/utils.ts

```diff
@@ -514,6 +514,22 @@ export function getUpdatedDataFromGrid(
             const row = editedRow.reduce((row, value, key) => {
                 const originalValue = originalRow.has(key) ? originalRow.get(key) : undefined;
 
+                // Convert empty cell to null
+                if (value === '')
```

Comment:
There are quite a few new caveats here. Due to the subtlety of some of these behaviors I think we should add more unit tests to cover these cases.

Reply:
Good call, I added extra checks in the existing jest tests that test booleans, ints, empty string, and string convert to boolean/number